internal/iam/repository_role.go

```diff
@@ -389,3 +390,44 @@ func (r *Repository) estimatedRoleCount(ctx context.Context) (int, error) {
 	}
 	return count, nil
 }
+
+// getRoleScopeType returns scope.Type of the roleId by reading it from the base type iam_role table
+// use this to get scope ID to determine which of the role subtype tables to operate on
+func getRoleScopeType(ctx context.Context, r db.Reader, roleId string) (scope.Type, error) {
+	const op = "iam.getRoleScopeType"
+	if roleId == "" {
+		return scope.Unknown, errors.New(ctx, errors.InvalidParameter, op, "missing role id")
+	}
+	if r == nil {
+		return scope.Unknown, errors.New(ctx, errors.InvalidParameter, op, "missing db.Reader")
+	}
+	rows, err := r.Query(ctx, scopeIdFromRoleIdQuery, []any{sql.Named("public_id", roleId)})
+	if err != nil {
+		return scope.Unknown, errors.Wrap(ctx, err, op, errors.WithMsg(fmt.Sprintf("failed to lookup role scope for :%s", roleId)))
+	}
+	var scopeId string
+	cnt := 0
+	for rows.Next() {
+		cnt++
+		if err := r.ScanRows(ctx, rows, &scopeId); err != nil {
+			return scope.Unknown, errors.Wrap(ctx, err, op, errors.WithMsg(fmt.Sprintf("failed scan results from querying role scope for :%s", roleId)))
+		}
```

Comment:
I'd only expect one value to be returned for `scopeId`, but just incase there isn't, what if we added an extra guard around that, like:

```suggestion
		}
        if cnt > 1 {
			return scope.Unknown, errors.New(ctx, errors.MultipleRecords, op, fmt.Sprintf("expected 1 row but got: %d", cnt))
		}
```